include/fc/container/flat.hpp

```diff
@@ -86,11 +86,16 @@ namespace fc {
           --_max_depth;
           unsigned_int size;
           unpack( s, size, _max_depth );
-          value.resize( size );
           if( !std::is_fundamental<T>::value ) {
-             for( auto& item : value )
-                unpack( s, item, _max_depth );
+             value.resize( std::min( size.value, FC_MAX_PREALLOC_SIZE ) );
+             for( uint64_t i = 0; i < size; i++ )
+             {
+                if( i >= value.size() )
```

Comment:
Here can probably be optimized to not compare `i` with `value.size()` every time. E.G.

```
max_size = FC_MAX_PREALLOC_SIZE;
for( i = 0; i < size; )
{
   if( i > 0 && max_size < size )
      max_size *= 2;
   value.resize( min( size, max_size ) );
   while( i < value.size() )
   {
      unpack(i);
      ++i;
   }
}
```

I'm not sure if it worth the efforts though.

Reply:
I don't think it's worth it. My guess would be that a typical implementation of `vector::size` will directly return a member variable.